1/src/Shape.ts

```diff
@@ -0,0 +1,43 @@
+import {Point} from "./point";
+
+export class Shape {
+  public color: string;
+  public filled: boolean;
+  public points: Point[];
+
+  constructor(points: Point[]);
+  constructor(points: Point[], color: string, filled: boolean);
+  constructor(points: Point[], color?: string, filled?: boolean) {
+    if (points.length < 3) {
+      throw new Error('Should be at least 3 points');
+    }
+    this.points = points;
+
+    if (!color && !filled) {
+      this.color = 'green';
+      this.filled = true;
+    } else {
+      this.color = color;
+      this.filled = filled;
+    }
+  }
+
+  public toString(): string {
+    const resultString: string = `A Shape with color of ${this.color} and ${this.filled ? 'filled' : 'not filled'}. Points:`
```

Comment:
No need to implicitly specify type here. Typescript will understand this as `string` by default. 
https://dev.to/ahmeddammarr/implicit-vs-explicit-types-in-typescript-1i2a